common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -239,9 +239,17 @@ private void captureGitMetadata() {
             String gitRepo = execAndGetStdOut("git", "config", "--get", "remote.origin.url");
             String gitCommitId = execAndGetStdOut("git", "rev-parse", "--verify", "HEAD");
             String gitCommitShortId = execAndGetStdOut("git", "rev-parse", "--short=8", "--verify", "HEAD");
-            String gitBranchName = execAndGetStdOut("git", "rev-parse", "--abbrev-ref", "HEAD");
             String gitStatus = execAndGetStdOut("git", "status", "--porcelain");
 
+            String gitBranchName;
+            // It is okay to call System.getenv directly here since this lambda is saved to the configuration cache
+            // and re-executed on every build.
+            if (isNotEmpty(System.getenv("BRANCH_NAME"))) {
```

Comment:
should we store this as a string on line 243 to be consistent with the other variables?

Reply:
I believe the latest commit addresses this.